tools/build-flow-types.js

```diff
@@ -0,0 +1,155 @@
+const path = require("path");
+const ancesdir = require("ancesdir");
+
+/**
+ * Let's use async/await to make life nicer.
+ */
+const readFileAsync = promisify(fs.readFile);
+const writeFileAsync = promisify(fs.writeFile);
+
+/**
+ * Some regex's we use to support our additional-files syntax.
+ */
+/**
+ * The first extracts the @additional-exports block, which looks like this:
+ *
+ * @additional-exports [
+ *    "./filea.js",
+ *    "./fileb.js",
+ * ]
+ *
+ * Paths in that block are assumed to be relative to the directory of the file
+ * declaring them.
+ *
+ * The second regex grabs the filenames themselves from this block.
+ */
+const regexAdditionalFiles = /^(?:\s*.*@additional-exports\s?\[)((?:\s*.*"(?:[./\-\w]+)",\s*)*)(?:\s*.*\])$/im;
+const regexExtractFile = /^(?:\s*.*"([./\-\w]+)",\s*)$/gim;
+
+/**
+ * Get the list of files that a given package exports.
+ */
+const getExportsFrom = async (
+    filePath /*: string*/,
+) /*: Promise<Array<string>>*/ => {
+    /**
+     * We assume the file we're given is an export as that's how our
+     * additional-exports syntax works.
+     */
+    const files = [path.basename(filePath)];
+    try {
+        const fileContent = await readFileAsync(filePath);
+
+        /**
+         * First regex gets the block of file names as indicated by
+         * the @additional-exports marker, between it's [] brackets.
+         * We don't need to loop the regex here as we only want the first
+         * match. We don't care if there are more.
+         *
+         * The list should have one file per line within double quotes
+         * and ending in a comma (went with the comma to make it feel like
+         * our coding standard).
+         */
+        const additionalFilesMatch = regexAdditionalFiles.exec(fileContent);
+        if (additionalFilesMatch != null && additionalFilesMatch[1] != null) {
+            /**
+             * We have a match, so let's now parse each file and add that to
+             * our list.
+             */
+            let additionalFileMatch;
+            do {
+                additionalFileMatch = regexExtractFile.exec(
+                    additionalFilesMatch[1],
+                );
+                if (
+                    additionalFileMatch != null &&
+                    additionalFileMatch[1] != null
+                ) {
+                    files.push(additionalFileMatch[1]);
+                }
+            } while (additionalFileMatch != null);
+        }
+        /**
+         * We will worry about duplicates later.
+         */
+        return files;
+    } catch (e) {
+        throw new Error(`Unable to process file: ${filePath} ${e}`);
+    }
+};
+
+/**
+ * Write out a .js.flow file in our disk for the file specified.
+ */
+const writeFlowFile = (
+    exportFromPath /*: string*/,
+    outputPath /*: string*/,
+) /*: Promise<void>*/ => {
+    const contents = ["// @flow", `export * from "${exportFromPath}";`].join(
+        "\n",
+    );
```

Comment:
Will this work with a module that does a default export?

Reply:
I copied it from Wonder Blocks. I sure hope so